apis/rust/node/src/node/mod.rs

```diff
@@ -401,7 +401,7 @@ impl Drop for DoraNode {
                 );
             }
 
-            match self.drop_stream.recv_timeout(Duration::from_secs(10)) {
+            match self.drop_stream.recv_timeout(Duration::from_millis(500)) {
```

Comment:
It might be but this only happen on a Node Drop. In the context of stopping a dataflow, so even though this might create read error, it is only in the context of dropping nodes.

Also, I think that dropping node should be fast and having to wait 10 seconds is never something that should be acceptable.

Reply:
It would also happen when a node finishes normally and the other nodes keep running, wouldn't it?

I agree that a 10s _in the drop handler_ is too long. But we should still wait the 10s before exiting the process and removing the shared memory that might still be needed by receiving nodes. Alternatively, we could to transfer ownership of the shared memory region to e.g. the deamon to enable a node exit without wait.